### ISSUE TITLE
Selltile doesn't work with some terminals (Gnome Terminal included) on GNOME 3.38.1

### ISSUE THREAD

**Issue by jeremie2** [emasab/shelltile#118]
**GNOME 3.38.1
Ubuntu 20.10**

I've tried to install different terminals to see what happens. In some case Shelltile simply doesn't work, in other cases it works if the terminal has 2 tabs open. Some other terminals has no problems at all.

**TERMINALS THAT DON'T WORK**
Nothing happens dragging window to sides or corners except when the window is dragged to the top side. In that case the window will be maximized.
This happens with the following terminals:
- Gnome Terminal
- LXTerminal
- Xfce4-terminla

**TERMINALS THAT WORK WITH 2 OPEN TABS**
- Terminology (default terminal on Enlightenment DE)
- ehm... at least another terminal had the same behaviour, but I've uninstalled it and can't remember which was... so sorry :P

**TERMINALS THAT WORKS**
- Sakura
- Konsole (default KDE terminal)
- Terminator

**Comment by giles-carre** [comment]
Hi Emanuele,

First of all, happy new year 2021.

I have the same kind of problem, only with terminal app, since upgrade from Ubuntu 20.04 to Ubuntu 20.10 (reinstall from scratch, including all my dot configuration home files).

I use only Gnome Terminal (v 3.38.0) with Gnome 3.38.1, and ShellTile version 64.

The problem occurs only with terminal (it's OK for all other apps). The behaviour is :

- the term resizes correctly : for exemple, if drag it to the bottom left corner, it resizes to the 1/4 of the screen
- but it stays in place, ie the head bar just above the bottom of the screen.

I have tried some tests :

- work with only one screen
- restore gnome term default configurations

but the problem persists.

If you want, I can do other tests.

And, thank you again for this wonderful extension.

Giles Carré
INSA Toulouse - France
giles.carre@insa-toulouse.fr

**Comment by axtimwalde** [comment]
Same problem here.  Everything works as expected except for the terminals.

**Comment by jeremie2** [comment]
A quick update,
Gnome Terminal is having the same problems also on **Ubuntu 20.04** with **Gnome 3.36.8**, probably because of some new updates.

I didn't try any other terminals.

**Comment by giles-carre** [comment]
Breaking news : same problem with Signal for Linux (Signal Desktop v1.39.5)

**Comment by axtimwalde** [comment]
And one more: Inkscape 1.0.1 (0767f8302a, 2020-10-17) installed via snap.

**Comment by nlflint** [comment]
Having same problem with Gnome shell on Ubuntu 20.04: snapping the terminal resizes the window correctly, but just doesn't put it in the right location (it's mostly off screen). This started right after rebooting from a bunch of auto-updated packages today. Good news, Alacrity terminal is working fine.

**Comment by emasab** [comment]
Hi, sorry for the wait. @jeremie2 Could you confirm that you're using Xorg and not Wayland, for example with
`echo $XDG_SESSION_TYPE`

**Comment by jeremie2** [comment]
@emasab
Yes it's Xorg session, sorry forgot to mention it! I'm stuck with Xorg because Middle-clicking title bars does not lower the window under Wayland. 

```
$ echo $XDG_SESSION_TYPE
x11
```

I'm almost sure it,s **off topic** but, just incase, this is the situation under Wayland:
ShellTile by default works only with keyboard shortcuts `ctrl + super + arrows` (.. even Gnome Terminal works).

It happened that ShellTile started to work with mouse gestures after I have tried other tiling window extensions which work under Wayland ([WinTile](https://extensions.gnome.org/extension/1723/wintile-windows-10-window-tiling-for-gnome/) and [Keypad Tiling](https://extensions.gnome.org/extension/2473/keypad-tiling/)).
Unfortunately this happens ramdomly. I couldn't find a way to reproduce the steps that make ShellTile work under Wayland.

Maybe a look inside WinTile source code could help..

**Comment by emasab** [comment]
On Wayland it won't work, if something works is by chance. The reason behind this is explained here https://github.com/emasab/shelltile/issues/98. I'm working on Wayland support and after the rewrite will test this behavior on both display servers

**Comment by giles-carre** [comment]
@emasab
Hi Emanuele,

For me, it's X Org too. Wayland isn't installed on my Ubuntu 20.10.

@Nathan : Alacrity doesn't work at all for me (it doesn't start).

Giles

**Comment by emasab** [comment]
@giles-carre  yes, it's a bug I've reproduced on Gnome 3.38. Strangely, with this new version, if you do
```
this.meta_window.move_resize_frame(true, x, y, w, h);
```
it only resizes but doesn't move the window for some kinds of windows. If you do a
```
this.meta_window.move_frame(true, x, y);
```
afterwards, then it works.

Try the latest version on master, it should be fixed now. 

Unfortunately on Gnome 3.38 window grouping in the overlay is not working because they've changed the window placing code another time, but that's another issue.

**Comment by jeremie2** [comment]
@emasab 
Just tried the updated version, happy to confirm that Gnome Terminal is snapping as expected.

Thank you very much!

**Comment by giles-carre** [comment]
@emasab 

I have patched windows.js as your indications, and, marvelous, it works.

This extension is the better I use

Thanks a lot

**Comment by nlflint** [comment]
The `move_frame()` patch fixed the problem for me too. thanks!

**Comment by axtimwalde** [comment]
Just installed the update and now everything works as expected. Thanks so much!
